lib/downloadManager.js

```diff
@@ -52,77 +52,87 @@ class PackageInfo {
+
+    let userProxy = npmConf.get('http-proxy');
+    if (userProxy != undefined && userProxy != '') {
+        console.info('[webide-common-tizentv]getPackageInfo(): userProxy = ' + userProxy);
+        pkgList = {'url':rootURL + `/pkg_list_${platform}`, 'proxy': userProxy};
+    }
 
     return new Promise((resolve, reject) => {
-        console.info('[webide-common-tizentv]getPackageInfo(): request to: ' + pkgList);
+        console.info('[webide-common-tizentv]getPackageInfo(): request to: ');
+        console.info(pkgList);
         let req = request(pkgList, (error, response, body) => {
             if (!response || response.statusCode != 200) {
                 console.error('[webide-common-tizentv]getPackageInfo(): Failed to connect: ' + pkgList);
                 reject(new Error(`Failed to archive tools infomation. Code:[${response.statusCode}]`));
             }
         });
 
-				let pkgInfo = new PackageInfo(rootURL);
-				let rl = readline.createInterface({input: req});
-				rl.on('line', line => {
-						if (pkgInfo.archived) {
-								rl.close();
-						}
-
-						if (line.includes('Package : ' + packageName)) {
-								console.info('[webide-common-tizentv]getPackageInfo(): got package info: ' + line);
-								pkgInfo.archived = false;
-						}
-
-						if (pkgInfo.archived === false) {
-						    pkgInfo.collectPkgInfo(line);
-						}
-				});
-
-				rl.on('close', () => {
-						resolve(pkgInfo);
-				});
-		});
+        let pkgInfo = new PackageInfo(rootURL);
+        let rl = readline.createInterface({input: req});
+        rl.on('line', line => {
+            if (pkgInfo.archived) {
+                rl.close();
+            }
+
+            if (line.includes('Package : ' + packageName)) {
+                console.info('[webide-common-tizentv]getPackageInfo(): got package info: ' + line);
+                pkgInfo.archived = false;
+            }
+
+            if (pkgInfo.archived === false) {
+                pkgInfo.collectPkgInfo(line);
+            }
+        });
+
+        rl.on('close', () => {
+            resolve(pkgInfo);
+        });
+    });
 }
 
 async function downloadPkg(packageInfo) {
     console.info('[webide-common-tizentv]downloadPkg(): downloadPkg start...');
-		await downloadPkgFormPath(packageInfo.rootURL + packageInfo.path);
+    await downloadPkgFormPath(packageInfo.rootURL + packageInfo.path);
 }
 
 async function downloadPkgFormPath(packagePath) {
     console.info('[webide-common-tizentv]downloadPkg(): downloadPkgFormPath start...');
-
-    //let tmpDir = path.resolve(extensionRoot, 'tmp');
     if (!fs.existsSync(tmpDir)) {
         fs.mkdirSync(tmpDir);
     }
 
-		let pkgName = 'temp.zip';
-		let pathArr = packagePath.split('/');
-		pkgName = pathArr[pathArr.length - 1];
+    let pkgName = 'temp.zip';
+    let pathArr = packagePath.split('/');
+    pkgName = pathArr[pathArr.length - 1];
+    let requestOpt = packagePath;
+
+    let userProxy = npmConf.get('http-proxy');
```

Comment:
https://wesbos.com/let-vs-const Read this article. You may understand difference between `let` and `const`. Shortly, use `const` keyword if the variable is `object` or `array` type. You can change the property of `object` or `array`. 
``` js
const a = [1]
// a = [1]
a.push(2)
// a = [1, 2]
```
Immutable means block the assignment( `=` operator). And if you want to block change the property, than use `Object.freeze()`. https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze

Reply:
done